### DISCUSSION TITLE
InfluxDB: Support BigInt values

### DISCUSSION THREAD

**Discussion by gji** [grafana/grafana#38383] [Dashboards, Panels & Visualizations]
**What happened**:
Grafana graph breaks when getting int64:
Uncaught (in promise) TypeError: Cannot mix BigInt and other types, use explicit conversions
    at e.value (time_series2.ts:301)
Error is slightly different in Firefox, but of similar spirit. I think problem comes from Arrow returning a BigInt array in js hold the int64s as native int64s don't exist. Line 233 defines delta as zero:
`this.stats.delta = 0;`
If the values are BigInts, implicit type conversion won't happen on the line that raises an error.

**What you expected to happen**:
No error

**How to reproduce it (as minimally and precisely as possible)**:
Using influxdb flux datasource plugin and loading in a table that contains ints. Influxdb uses int64.

**Anything else we need to know?**:
I guess the quick way to fix this is to just cast column values to Number if they're BigInt.

**Environment**:
- Grafana version: 7.0
- Data source type & version: InfluxDB flux datasource v7.0.0, InfluxDB 2.0 beta 10
- OS Grafana is installed on: Docker
- User OS & Browser: Windows, Firefox and Chrome
- Grafana plugins:
- Others:

**Comment by gji** [comment]
I'm not sure this has been categorized correctly. While the issue I had was with the beta influxDB plugin, any backend plugin that returns int64 will have this issue.

In fact, you can simply modify the sample code from running grafana/toolkit to create a new backend plugin. Just modify sample-plugin.go to return several values (around line 90):
```
  // add the time dimension
  frame.Fields = append(frame.Fields,
    data.NewField("time", nil, []time.Time{
      query.TimeRange.From.Add(time.Second*0),
      query.TimeRange.From.Add(time.Second*1),
      query.TimeRange.From.Add(time.Second*2),
      query.TimeRange.From.Add(time.Second*3),
      query.TimeRange.From.Add(time.Second*4),
      query.TimeRange.From.Add(time.Second*5),
    }),
  )

  // add values
  frame.Fields = append(frame.Fields,
    data.NewField("values", nil, []int64{10, 20, 10, 20, 10, 50}),
  )
```
You can no longer see the plot, and the UI throws the same javascript error. Even without this modification, the fact that the tooltip hover doesn't show the "values" field might be another symptom of this problem.

**Comment by torkelo** [comment]
@kylebrandt @ryantxu could this be problem with data frame serialization m?

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
